### ISSUE TITLE
Port twisted.internet.posixbase to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6103]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6103|
|Type|enhancement|
|Created|2012-10-12 14:11:38Z|
|Branch|https://github.com/twisted/twisted/tree/posixbase-py3-6103-2|

`twisted.internet.posixbase` should run on Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__6103 6103
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_posixbase_py3_6103_2 branches/posixbase-py3-6103-2
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1350051098000000 1350051098000000
changetime__1350397615000000 1350397615000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [36123]) Branching to 'posixbase-py3-6103'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst commented|
|-|-|

Notes for when this is in review:

  1. This is only a limited port; obviously many of the methods (e.g. process support) are untested.
  1. I moved some tests, unchanged, from `test_posixbase` to the more appropriate `twisted.test.test_unix` and `twisted.test.test_tcp_internals` (I chose those because we're trying to move `twisted.internet.test` towards `ReactorBuilder`, with `twisted.test` being the old crufty stuff that needs rethinking/rewriting).

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst set owner to @exarkun|
|-|-|

Ready for review:
http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/posixbase-py3-6103

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @itamarst|
|-|-|

Thanks.

  1. The test organization was pretty bad before, but I'm not sure this move is a huge improvement.  Adding more code to the modules we're trying to get rid of is an obvious step backwards.  I think it's incorrect to say we're trying to move `twisted.internet.test` towards `ReactorBuilder`.  Rather, we're trying to move blackbox reactor testing towards `ReactorBuilder`, and we're trying to move tests for code in `twisted.internet` into `twisted.internet.test`.  I think the two choices we have now, `twisted.internet.test.test_posixbase` and `twisted.test.test_tcp`, are both pretty bad, and really a symptom of another problem: `twisted.internet.posixbase` should really be part of a new `twisted.internet` package, `posixreactor` or something - a sibling of `iocpreactor`.  Squishing it down into `twisted.internet` leads to confusion and poor naming.

Hmm.  And that's really all that's happening in this branch, so that's all I have to say about it I guess. :)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst commented|
|-|-|

(In [36166]) Branching to 'posixbase-py3-6103-2'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_6"></a>|@itamarst set owner to @exarkun|
|-|-|

OK, I moved tests back into `test_posixbase`.

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/posixbase-py3-6103-2 is/will be running.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @itamarst|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_8"></a>|@itamarst set status to `closed`|
|-|-|

(In [36184]) Merge posixbase-py3-6103-2.

Author: itamar
Review: exarkun
Fixes: [#6103](https://github.com/twisted/twisted/issues/6103)

Port most of twisted.internet.posixbase to Python 3 (Windows and process support are untested).
